src/main/java/io/github/syst3ms/skriptparser/expressions/ExecExprListOperators.java

```diff
@@ -90,53 +120,73 @@ public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContex
 		if (values.length == 0)
 			return new Object[0];
 
-		switch (pattern) {
+		switch (type) {
 			case 0:
 				list.change(ctx, Arrays.copyOfRange(values, 0, values.length - 1), ChangeMode.SET);
 				return new Object[] {values[values.length - 1]};
 			case 1:
 				list.change(ctx, Arrays.copyOfRange(values, 1, values.length), ChangeMode.SET);
 				return new Object[] {values[0]};
 			case 2:
-				var low = lower.getSingle(ctx);
-				if (low.isEmpty() || low.get().compareTo(BigInteger.ZERO) <= 0) {
-					list.change(ctx, new Object[0], ChangeMode.SET);
+				int ind = index.getSingle(ctx)
+						.filter(n -> n.signum() > 0 && n.compareTo(BigInteger.valueOf(values.length)) <= 0)
+						.map(n -> n.intValue() - 1)
+						.orElse(-1);
+				if (ind == -1) {
 					return new Object[0];
 				}
-				int b1 = low.get().intValue() - 1;
-				int b2 = upper != null
-						? upper.getSingle(ctx)
-								.filter(u -> u.compareTo(BigInteger.valueOf(values.length)) <= 0)
-								.map(u -> (BigInteger) u)
-								.orElse(BigInteger.valueOf(values.length)).intValue()
+				var skipped = new Object[values.length - 1];
+				for (int i = 0, k = 0; i < values.length; i++) {
+					if (i == ind) {
+						continue;
+					}
+					skipped[k++] = values[i];
+				}
+				list.change(ctx, skipped, ChangeMode.SET);
+				return new Object[] {values[ind]};
+			case 3:
+				var low = lower != null
+						? lower.getSingle(ctx).filter(n -> n.signum() > 0).map(n -> n.intValue() - 1).orElse(-1)
+						: 0;
+				var up = upper != null
+						? upper.getSingle(ctx).filter(n -> n.compareTo(BigInteger.valueOf(values.length)) <= 0).map(BigInteger::intValue).orElse(values.length)
 						: values.length;
-				if (b1 > b2) {
+				var st = step != null
+						? step.getSingle(ctx).filter(n -> n.signum() > 0 && n.compareTo(BigInteger.valueOf(values.length)) <= 0).map(BigInteger::intValue).orElse(-1)
+						: 1;
+
+				if (low == -1 || up == -1 || up == 0 || st == -1 || low > up) {
```

Comment:
A step can be negative. Using python syntax:
```python
a = [2,3,5,7,11,13]
a[5:1:-1] == [13,11,7,5]
```

Reply:
That sounds really odd to me and I don't like it as a feature, but it as said before: programmers can use their gimmicks and non-programmers can refer to the easier syntax.

Reply:
Having a non-1 step isn't that rare of a situation, and whether it's negative or not just depends on what ordering is most convenient for the implementation at hand, so it isn't *that* rare.
Step -1 is just a fancy way of saying "reversed", hence the idiomatic `list[::-1]` to reverse a Python list, but other negative steps require a bit more tricks to properly express.
For example, `{list::*} starting at {_n} with step -{_m}` is in fact not equal to `reversed ({list::*} up to {_n} with step {_m})`, but actually `reversed ({list::*} from {_n} % {_m} to {_n} with step {_m})`, which is not very clear.